### PR TITLE
[REVIEW] Change strings::split_record to return a lists column

### PR DESCRIPTION
**Pull request by davidwendt**
Reference #5667 

The `cudf::strings::contiguous_split_record` API returns a single memory buffer plus a vector of `column_view`s similar to the `cudf::contiguous_split` function. Now that we have LIST type columns, the strings API has been repurposed in this PR to return a lists column instead. The lists column child will be a flat strings column and the list offsets identify each input string's tokens.

Since the new APIs no longer matches the `contiguous_split` result, this PR also changes the names to simply `cudf::strings::split_record` and `cudf::strings::rsplit_record`. All appropriate gtests have been updated accordingly.

The code change was involved because the original layout of characters and offsets were interleaved in the output memory. The new result creates one large strings column of all the tokens along with the appropriate offsets for the lists column.

This API is not currently exposed in the cudf Python interface but could be used in the `str.split()` functions with `expand=False`

This PR does not include adding regex support to any of the existing `split()` APIs.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by harrism** [comment]
Does this close #5667, or just "reference" it?

**Comment by davidwendt** [comment]
> Does this close #5667, or just "reference" it?

This PR includes only the libcudf C++ code for this API and not any of the Python/Cython bindings to it.
The Lists support for Python cudf is ongoing so the reference issue will likely be closed in a future PR.

File: cpp/tests/strings/split_tests.cpp [hunk 1]
```diff
@@ -275,77 +275,44 @@ TEST_F(StringsSplitTest, AllNullsCase)
...
-  EXPECT_TRUE(result.column_views.size() == expected_columns.size());
-  for (size_t i = 0; i < result.column_views.size(); ++i) {
-    cudf::test::expect_columns_equal(result.column_views[i], *expected_columns[i]);
-  }
+  auto result =
+    cudf::strings::split_record(cudf::strings_column_view(strings), cudf::string_scalar(" "));
+  cudf::lists_column_view lcv(result->view());
+  cudf::test::strings_column_wrapper expected(
+    {"", "Héllo", "thesé", "are", "some", "", "", "tést", "String", ""});
+  cudf::test::fixed_width_column_wrapper<int32_t> offsets({0, 3, 3, 7, 9, 10});
```
Comment (revans2) [inline cpp/tests/strings/split_tests.cpp:291]: I thought that @nvdbaranec had APIs to create list columns for tests that would make this even smaller.  Do those not work for string columns?

https://github.com/rapidsai/cudf/blob/ce826c57c8cafb7d78f33873ca8c0c030fa42087/cpp/tests/utilities/column_wrapper.hpp#L587-L606

**Comment by revans2** [comment]
Looks good to me too.

**Pull request merged**